rir/src/interpreter/interp.cpp

```diff
@@ -1520,11 +1535,13 @@ SEXP evalRirCode(Code* c, InterpreterInstance* ctx, SEXP env,
 #endif
 
     assert(c->info.magic == CODE_MAGIC);
+    bool existingLocals = localsBase;
 
     BindingCache bindingCache[BINDING_CACHE_SIZE];
-    memset(&bindingCache, 0, sizeof(bindingCache));
+    bool smallCache = c->bindingsCount <= BINDING_CACHE_SIZE;
+    if (env != symbol::delayedEnv || existingLocals)
```

Comment:
I don't think existingLocals allows you to not initialize the cache

Reply:
can you add a comment that the zeroing is delayed to the setEnv, it took me a bit to understand it